### PR TITLE
RCTRefreshControl: Updates progressOffset behaviour to prevent it from being applied by default

### PR DESCRIPTION
**Pull request by objectivecosta**
## Summary

UIRefreshControl has a tight integration with iOS in terms of UINavigationBar/UIScrollView. In this regard, whenever there's a UIScrollView with a UINavigationBar on top, the OS automatically adjusts the contentInset of the UIScrollView to reflect that (something that is exposed to React Native). In a similar manner, UIScrollView passes this along to the attached UIRefreshControl.

By setting the frame manually, the RCTRefreshControl component was preventing this behavior. Although having the option is desired, it should not be done by default. In the past it was possible to adjust for this by manually setting the correct value, calculating the statusBar's height/safeAreaInsets.top and appending 44pt (the UINavigationBar height). However, due to changes related to the Dynamic Island (see [here](https://useyourloaf.com/blog/iphone-14-screen-sizes)), the safe area and the status bar size no longer align, making this calculation more tricky.

In summary: this changes allows `progressViewOffset` to exist (in order to maintain feature parity with Android) but provides the opportunity for the OSs default behavior to kick in when applicable.

| Applying by default  | Not applying by default (this change) |
:-------------------------:|:-------------------------:
  |  

## Changelog

[iOS] [Fixed] - Fix application of _progressViewOffset in RCTRefreshControl to not occur by default (when value is unset)

## Test Plan

The GIFs attached display the behavior as expected/unexpected. I'm unaware of any tests written for RCTRefreshControl that could be improved to cover this change. Notes appreciated.

### CODE REVIEW

**Comment by objectivecosta** [comment]
One thing that I reflected upon is that maybe the condition can be changed for `!= 0.f` instead of `> 0.f`, to allow for negative offsets. Is that a possible scenario?

**Comment by objectivecosta** [comment]
Updated the condition to be `!= 0.f`, to allow for negative offsets. PR is ready for review, but I don't know how to request one here :/

**Review by cipolleschi — APPROVED** [review]
Hi @objectivecosta, thank you so much for doing this, that's great.

I added a comment to improve the readability of the code, but it is a personal preference (coming from Swift and `guard` statements) so feel free to disregard it if you don't feel like it.

I'll also look into why you couldn't request a review, but it's fine to ping me (hoping I don't miss the notification! 😅)

File: React/Views/RefreshControl/RCTRefreshControl.m [hunk 1]
```diff
@@ -121,11 +121,17 @@ - (void)_applyProgressViewOffset
...
-  UIView *scrollView = self.superview;
-  UIView *target = scrollView.superview;
-  CGPoint rawOffset = CGPointMake(0, _progressViewOffset);
-  CGPoint converted = [self convertPoint:rawOffset fromView:target];
-  self.frame = CGRectOffset(self.frame, 0, converted.y);
+
+  // Setting the UIRefreshControl's frame breaks integration with ContentInset from the superview
+  // if it is a UIScrollView. This integration happens when setting the UIScrollView's .refreshControl
+  // property. For this reason, setting the frame manually should be avoided, if not needed.
+  if (_progressViewOffset != 0.f) {
```
[the remark this answers is not in the document]
Comment (objectivecosta) [reply React/Views/RefreshControl/RCTRefreshControl.m:128]: Makes total sense @cipolleschi! Changed it ;)

**Comment by cipolleschi** [comment]
It has landed! https://github.com/facebook/react-native/commit/0062b10b56985c4556011fbbb8d43f0a038d359e

**Comment by objectivecosta** [comment]
Great stuff! Thanks @cipolleschi :D
